leshan-core/src/main/java/org/eclipse/leshan/core/LwM2m.java

```diff
@@ -91,35 +91,35 @@ public boolean equals(Object obj) {
      */
     public class Version implements Comparable<Version> {
 
-        public static Version V1_0 = new Version("1.0");
-        public static final Version MAX = new Version(Short.MAX_VALUE, Short.MIN_VALUE);
+        public static final Version V1_0 = new Version("1.0");
+        public static final Version MAX = new Version(Short.MAX_VALUE, Short.MAX_VALUE);
 
-        protected final Short major;
-        protected final Short minor;
+        protected final short major;
+        protected final short minor;
 
         public Version(int major, int minor) {
-            this.major = (short) major;
-            this.minor = (short) minor;
+            this((short) major, (short) minor);
+            if (this.major != major) {
+                throw new ArithmeticException("short overflow");
+            }
+            if (this.minor != minor) {
+                throw new ArithmeticException("short overflow");
+            }
         }
 
-        public Version(Short major, Short minor) {
+        public Version(short major, short minor) {
             this.major = major;
             this.minor = minor;
```

Comment:
should we add test to check that short is not negative ?

- `"version (%d.%d) major part %d must not be negative"`
- `"version (%d.%d) major part %d must not be negative"`

There is an issue with that, we will not be able to call `this((short) major, (short) minor);` in constructor above because error message could be wrong. I guess the code should be adapted a little more than just adding the check.

Any opinion about this ?

Reply:
Good point. I've added the negativity check there and refactored the int-to-short casting to include the correct error message.